bundles/specmate-migration-test/src/com/specmate/migration/test/ChangedTypesTest.java

```diff
@@ -0,0 +1,107 @@
+package com.specmate.migration.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull; 
+import static org.junit.Assert.assertTrue;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.junit.Test;
+
+import com.specmate.migration.test.changedtypes.testmodel.artefact.File;
+import com.specmate.migration.test.changedtypes.testmodel.base.BasePackage;
+import com.specmate.migration.test.changedtypes.testmodel.base.Folder;
+import com.specmate.model.support.util.SpecmateEcoreUtil;
+import com.specmate.persistency.ITransaction;
+
+public class ChangedTypesTest extends MigrationTestBase {
+
+	public ChangedTypesTest() throws Exception {
+		super("changetypes");
+	}
+	
+	@Test 
+	public void testNeedsMigration() throws Exception {
+		activatePersistency();
+		assertFalse(migratorService.needsMigration());
+		configureTestModel(BasePackage.class.getName());
+		assertTrue(migratorService.needsMigration());
+		deactivatePersistency();
+	}
+	
+	@Test
+	public void doMigration() throws Exception {
+		checkMigrationPreconditions();
+		
+		configureTestModel(BasePackage.class.getName());
+		// Once we know how to call the standard activate method of the persistency service, 
+		// we do not need to initiate the migration here as it is already initiated in the activate method
+		migratorService.doMigration();  
+		
+		checkMigrationPostconditions();
+	}
+	
+	private void checkMigrationPostconditions() throws Exception {	
+		activatePersistency();
+		
+		ITransaction transaction = persistencyService.openTransaction();
+		Resource resource = transaction.getResource();
+		EObject root = SpecmateEcoreUtil.getEObjectWithId("root", resource.getContents());
+		assertNotNull(root);
+		
+		assertTrue(root instanceof Folder);
+		Folder rootFolder = (Folder) root;
+		
+		EObject file = SpecmateEcoreUtil.getEObjectWithId("f0", rootFolder.eContents());
+		assertNotNull(file);
+		assertTrue(file instanceof File);
+		File f0 = (File) file;
+		
+		/*
+		 * Contrary to the documentation [1], variables of type byte are *not* stored as 
+		 * TINYINT, but as SMALLINT. Hence we omit here the test for the
+		 * conversion of type byte to X. 
+		 * 
+		 * [1] http://www.h2database.com/html/datatypes.html#tinyint_type
+		 */
+		
+		// Short to X conversion
+		assertEquals(3, f0.getShortVar1());
```

Comment:
Can you add a comment here to explain the rationale behind the checks?
My interpretation is: In the updated datamodel, shortvar1 is an int, shortvar2 is a long and so on. Is this correct?